include/lsst/afw/detection/Footprint.h

```diff
@@ -305,32 +206,26 @@ class Footprint : public lsst::daf::base::Citizen,
      *                       NOT the same as the footprint's bounding box.
      *  @param[in]  doClip   If true, clip the new footprint to the region bbox before returning it.
      */
-    PTR(Footprint) transform(
+    std::shared_ptr<Footprint> transform(
         image::Wcs const & source,
         image::Wcs const & target,
         geom::Box2I const & region,
         bool doClip=true
     ) const;
 
-    /**
-     * @brief Find edge pixels on the footprint
-     *
-     * Note that the resultant Footprint of edge pixels may not be contiguous.
-     */
-    PTR(Footprint) findEdgePixels() const;
+    bool isPersistable() const { return true; }
 
-    /**
-     *  @brief Update the Footprint in-place to be the union of itself and all others provided
-     *
-     *  Only spans will be modified; peaks will be left unchanged.  If ignoreSelf is true it
-     *  will only be the union of all its children.
-     *
-     */
-    void include(std::vector<PTR(Footprint)> const & children, bool ignoreSelf=false);
+    std::shared_ptr<Footprint> intersect(std::shared_ptr<Footprint> other) const;
+    std::shared_ptr<Footprint> intersectNot(std::shared_ptr<Footprint> other) const;
+    std::shared_ptr<Footprint> union_(std::shared_ptr<Footprint> other) const;
```

Comment:
I believe we discussed adding versions of these that take `(Mask<T> const & mask, T bitmask)`.